### ISSUE TITLE
[Suggestion] Biome blend for GUIs/entities

### ISSUE THREAD

**Issue by ZwipZwapZapony** [sp614x/optifine#2354]
A leaf block appears as a specific shade of green* when in inventory GUIs, held/worn by entities, dropped on the ground, et cetera. However, when placed, a leaf block takes on different shades of green*, depending on the biome that it's in and stuff like that. (* The "green" shades can be changed by resource packs.)

I'd like to suggest an option that allows that biome blend ("tintindex") to be applied even in inventory GUIs, as an entity, et cetera. While I'm suggesting for this specifically for block items (e.g. leaf blocks, grass blocks...), it might be nice to allow it for other entities (e.g. Zombies) as well through resource packs (for example, this could allow changing the blue parts of Drowneds to match the water of their current biome).

For GUI/inventory icons, an item's biome blend should probably be based on the biome/location of the local player. For entities, perhaps a "fancy" option would link it to the biome/location of the entity (or its wearer, if worn/held by another entity (e.g. in a player hand)), and a "fast" option would simply always link it to the biome/location of the local player.

~

TL;DR: Placed blocks can have biome blends ("tintindex") applied. I'd like to see that also applied in GUI inventory icons, dropped items, other entities, et cetera.

**Comment by sp614x** [comment]
Having the handheld block use the curent biome color would look nice.
Not sure about the inventory, the colors there should be static.
